internal/service/device.go

```diff
@@ -212,3 +214,57 @@ func (h *ServiceHandler) GetRenderedDeviceSpec(ctx context.Context, request serv
 		return nil, err
 	}
 }
+
+// (PATCH /api/v1/devices/{name})
+func (h *ServiceHandler) PatchDevice(ctx context.Context, request server.PatchDeviceRequestObject) (server.PatchDeviceResponseObject, error) {
+	orgId := store.NullOrgId
+
+	currentObj, err := h.store.Device().Get(ctx, orgId, request.Name)
+	if err != nil {
+		switch err {
+		case flterrors.ErrResourceIsNil:
+			return server.PatchDevice400JSONResponse{Message: err.Error()}, nil
+		case flterrors.ErrResourceNameIsNil:
+			return server.PatchDevice400JSONResponse{Message: err.Error()}, nil
+		case flterrors.ErrResourceNotFound:
+			return server.PatchDevice404JSONResponse{}, nil
+		default:
+			return nil, err
+		}
+	}
+
+	newObj := &v1alpha1.Device{}
+	err = ApplyJSONPatch(ctx, currentObj, newObj, *request.Body, "/api/v1/devices/"+request.Name)
+	if err != nil {
+		return server.PatchDevice400JSONResponse{Message: err.Error()}, nil
+	}
+
+	if newObj.Metadata.Name == nil || *currentObj.Metadata.Name != *newObj.Metadata.Name {
+		return server.PatchDevice400JSONResponse{Message: "metadata.name is immutable"}, nil
+	}
+	if currentObj.ApiVersion != newObj.ApiVersion {
+		return server.PatchDevice400JSONResponse{Message: "apiVersion is immutable"}, nil
+	}
+	if currentObj.Kind != newObj.Kind {
+		return server.PatchDevice400JSONResponse{Message: "kind is immutable"}, nil
+	}
+	if !reflect.DeepEqual(currentObj.Status, newObj.Status) {
```

Comment:
this is tricky because we have time stamps here, will that cause problems?

Reply:
It should not AFAIK, all the types are just string/int or some other primitives so `reflect.DeepEqual` will work for these.

Reply:
it should not though. If it has, it means that patch request changed it and it should be rejected. Or maybe I am missing something :)